backend/Auth/ResetPassword.ts

```diff
@@ -115,6 +119,63 @@ router.get<Params, ResponseBody, QueryParams>(
   }
 );
 
+router.put(
+  "/api/passwords",
+  body("password").isStrongPassword(),
+  body("token").exists(),
+  async (req, res) => {
+    const { password, token } = req.body;
+
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      if (token) {
+        return invalidRequest(
+          res,
+          "Password must be at least 8 characters long, contain one number, one uppercase letter, and one special character."
+        );
+      } else {
+        return invalidRequest(
+          res,
+          "Request must include token. Please reset your password again."
+        );
+      }
+    }
+
+    if (!tokenIsValid(token, jwtSecret)) {
```

Comment:
Does this validate expiration date? I think the `verify()` function might do that, but am not sure. If it doesn't, then we need to make sure we don't let them use an expired token.

Reply:
Confirmed verify() validates expiration date.